### DISCUSSION TITLE
Some TypeScript type hint missing when using Bun

### DISCUSSION THREAD

**Discussion by joshuaavalon** [windmill-labs/windmill#4283] [Questions]
I found there are 2 cases the web editor does not have type hint but the code can still be run. (Bun)

1. [package.json subpath exports](https://nodejs.org/api/packages.html#subpath-exports)
The web editor seems to referencing `types` in `package.json` but ignoring `exports`. So, if the package has subpath exports like `package-a/b`, the web editor cannot resolve.

2. TypeScript `extends`
```ts
class A {
  public a(): void {}
}

class B extends A {
  public b(): void {}
}

const b = new B();
b.a(); // a not exists on b
```

**Comment by rubenfiszel** [comment]
For the case 2), it seems to work as far as I can see. For the case 1), we will investigate but it might be the limit of what our auto-type fetcher can do

**Comment by joshuaavalon** [comment]
@rubenfiszel For case 2, I can confirm it is not working on the cloud.

```ts
import { initMdh } from "@media-data-hub/sdk";

async function main() {
  const a = await initMdh({});
  a.create();
}
```
![image](https://github.com/user-attachments/assets/c24fec7e-9989-42f5-8f1d-b77582677797)
![image](https://github.com/user-attachments/assets/1817d3e2-b4ab-4e76-91ef-26b12dcf1967)

`initMdh` return `MediaDataHub` which extends `PocketBase`. I think it may be related `PocketBase` is from another package (`pocketbase`).
